test/cql-pytest/test_tablets.py

```diff
@@ -107,3 +107,17 @@ def test_describe_initial_tablets(cql, skip_without_tablets):
     with new_test_keyspace(cql, ksdef) as keyspace:
         desc = cql.execute(f"DESCRIBE KEYSPACE {keyspace}")
         assert "and tablets = {'initial': 1}" in desc.one().create_statement.lower()
+
+
+def test_tablets_are_dropped_when_dropping_table(cql, skip_without_tablets):
+    ksdef = "WITH REPLICATION = {'class' : 'NetworkTopologyStrategy', 'replication_factor' : '1'} AND TABLETS = {'initial': 1}"
+    with new_test_keyspace(cql, ksdef) as keyspace:
+        keyspace_name = keyspace
+        table_name = ""
+        with new_test_table(cql, keyspace, "pk int PRIMARY KEY, c int") as table:
+            table_name = table.split('.')[1]
+            res = cql.execute(f"SELECT * FROM system.tablets WHERE keyspace_name='{keyspace_name}' AND table_name='{table_name}' ALLOW FILTERING")
+            assert res is not None, f"table {keyspace_name}.{table_name} not found in system.tablets after the table was created"
```

Comment:
Are you sure this is the right assert? I would have assumed that if there are zero results, you still get some object (not "None") which has the empty list of results?

Note that below you used "not res", not "res is None" - those are not the same thing.

Reply:
fixed in next version